src/Controller/ImportController.php

```diff
@@ -67,30 +82,347 @@ private function importDomainName(UploadedFile $csvFile, EntityManagerInterface
+     * @return void
+     * @throws \Exception
+     */
     private function importSite(UploadedFile $csvFile, EntityManagerInterface $entityManager) {
+        if ($csvFile && false !== ($fps = fopen($csvFile->getPath() . '/' . $csvFile->getFilename(), "r"))) {
+            $indexCSV = 0;
+            while (false !== ($sRow = fgetcsv($fps, 1000, ';'))) {
+                ++$indexCSV;
+                if (1 < $indexCSV) {
+                    $site = [
+                        'name' => $sRow[0],
+                        'onlineDate' => new \DateTime($sRow[1]),
+                        'online' => $sRow[2],
+                    ];
+
+                    $futurSite = new Site();
+                    $siteForm = $this->createForm(SiteType::class, $futurSite);
+                    $siteForm->submit($site);
+                    $siteForm->getData()->setOnlineDate($site['onlineDate']);
+
+                    $siteForm->getData()->setClient($entityManager->getRepository(Client::class)->findOneBy(['name' => $sRow[3]]));
+                    $siteForm->getData()->addServer($entityManager->getRepository(Server::class)->findOneBy(['name' => $sRow[4]]));
+
+                    $entityManager->persist($siteForm->getData());
+                }
+            }
+            $entityManager->flush();
+        }
+    }
+
+    /**
+     * @param UploadedFile $csvFile
+     * @param EntityManagerInterface $entityManager
+     * @return void
+     * @throws \Exception
+     */
+    private function importClickAndCollect(UploadedFile $csvFile, EntityManagerInterface $entityManager) {
+
+        if ($csvFile && false !== ($fps = fopen($csvFile->getPath() . '/' . $csvFile->getFilename(), "r"))) {
+            $indexCSV = 0;
+            while (false !== ($sRow = fgetcsv($fps, 1000, ';'))) {
+                ++$indexCSV;
+                if (1 <= $indexCSV) {
+                    $clickAndCollect = [
+                        'provider' => $sRow[0],
+                        'offer' => $sRow[1],
+                        'invoicedPrice' => $sRow[2],
+                        'cost' => $sRow[3],
+                        'renewalType' => $sRow[4],
+                        'startDate' => new \DateTime($sRow[5]),
+                        'renewalDate' => new \DateTime($sRow[6]),
+                        'onlineDate' => new \DateTime($sRow[7]),
+                        'online' => $sRow[8],
+                    ];
+                    if ("" === $clickAndCollect['renewalDate']) {
+                        $clickAndCollect['renewalDate'] = date_add(new \DateTime($sRow[5]) , new DateInterval('P1Y'));
+                    }
+
+                    $futurClickAndCollect = new ClickAndCollect();
+                    $clickAndCollectForm = $this->createForm(ClickAndCollectType::class, $futurClickAndCollect);
+                    $clickAndCollectForm->submit($clickAndCollect);
+                    $clickAndCollectForm->getData()->setStartDate($clickAndCollect['startDate']);
```

Comment:
Same here : store $clickAndCollectForm->getData in a var and use it with fluent setters